rp2040-hal/src/usb.rs

```diff
@@ -54,6 +53,56 @@
 //!
 //! If the required timing cannot be met, using an maximum packet size of the endpoint 0 above 18bytes
 //! (e.g. `.max_packet_size_ep0(64)`) should avoid that issue.
+//!
+//! ## Issue on RP2040B0 and RP2040B1: USB device fails to exit RESET state on busy USB bus.
+//!
+//! The feature `rp2040-e5`implements the workaround described by [RP2040-E5](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#%5B%7B%22num%22%3A630%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C115%2C158.848%2Cnull%5D).
+//!
+//! The workaround requires the GPIO block to be released from its reset and has for side effect
+//! that GPIO15 will be stolen for a few hundred microseconds each time a Reset is detected on the
+//! USB bus.
+//!
+//! Although the external state of the pin will not be affected (the pin it put in "bus keep"), a
```

Comment:
I don't think this is exactly true: If the pin was configured as an active output, the external state will switch from "actively driven" to "weak pull-up/down".

Reply:
TBH, I'm don't know enough about HW to easily check that.
I tried to find a reference to "bus keep" in the datasheet in vain (I saw that feature on FPGA design but I'm not sure about what it does).

I based that comment on the reference implementation in the [pico-sdk](https://github.com/raspberrypi/pico-sdk/blob/afc10f3599c27147a6f34781b7102d86f58aa5f6/src/rp2_common/pico_fix/rp2040_usb_device_enumeration/rp2040_usb_device_enumeration.c#L97-L98).

Reply:
Yes, that matches my understanding. There is surprisingly little documentation, it seems like the bus-keep feature is not mentioned in the datasheet at all.
The [SDK](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_gpio/include/hardware/gpio.h#L203-L204) contains a comment "On the RP2040, setting both pulls enables a "bus keep" function, i.e. a weak pull to whatever is current high/low state of GPIO.", that's all the documentation I can find. But still, a "weak pull" is different from an actively driven output.

Reply:
A thing I don't know is if the bus is currently high and something comes to pull it low.
Is the pin still driven high by the bus keep or does it switch?

I'll try (I hope there won't be magic smoke)

Reply:
It should switch. The general idea of such a bus keep feature is described here: https://en.wikipedia.org/wiki/Bus-holder
(The actual implementation on the rp2040 might be different, but the result should be the same.)

Reply:
It does indeed :) so the comment seems to be correct :D

Reply:
Only as long as nothing is connected to the pin - which seems like a rather uninteresting situation.
But for example, if you have  an external pull-up resistor, it is probably much stronger than the bus-keep mechanism. So if the pin was configured as an active output with level low, it will temporarily switch to high during the e5-workaround.

I'd prefer a wording like:
"The pin will be temporarily put in "bus keep" mode, weakly pulling the
output towards its current logic level. In absence of external loads,
the current logic level will be maintained.
A user will lose control of the pin's output and reading from it may
not reflect the actual state of the external pin."